### ISSUE TITLE
Protolint  substituting "message" with "  sage"

### ISSUE THREAD

**Issue by navaneeth-spotnana** [yoheimuta/protolint#139]
I am running `protolint version 0.26.1(84c57fd)` running on `macOS 11.0.1`, and it's making this wrong transformation:

![image](https://user-images.githubusercontent.com/57885431/101182456-1caae480-3674-11eb-9d66-d5d4077bd317.png)

Looks like it's making a wrong transformation. To reporoduce the issue, save the below text into a .proto file:

```
syntax = "proto3";

package foobar;

option java_package = "com.test.foo.bar";

message TestMessage { string test_field = 1; }
```

Run:
```
protolint lint -v -fix test.proto
```

It prints the below output to console:
```
2020/12/04 21:05:04 [DEBUG] Text=[syntax], Token=[2], Pos=[test.proto:1:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[syntax], Token=[22], Pos=[test.proto:1:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[=], Token=[10], Pos=[test.proto:1:8] called from syntax.go:63
2020/12/04 21:05:04 [DEBUG] Text=["], Token=[11], Pos=[test.proto:1:10] called from syntax.go:68
2020/12/04 21:05:04 [DEBUG] Text=[proto3], Token=[2], Pos=[test.proto:1:11] called from syntax.go:73
2020/12/04 21:05:04 [DEBUG] Text=["], Token=[11], Pos=[test.proto:1:17] called from syntax.go:79
2020/12/04 21:05:04 [DEBUG] Text=[;], Token=[8], Pos=[test.proto:1:18] called from syntax.go:84
2020/12/04 21:05:04 [DEBUG] Text=[package], Token=[2], Pos=[test.proto:3:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[package], Token=[2], Pos=[test.proto:3:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[package], Token=[2], Pos=[test.proto:3:1] called from parser.go:44
2020/12/04 21:05:04 [DEBUG] Text=[package], Token=[29], Pos=[test.proto:3:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[package], Token=[29], Pos=[test.proto:3:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[foobar], Token=[2], Pos=[test.proto:3:9] called from fullIdent.go:8
2020/12/04 21:05:04 [DEBUG] Text=[;], Token=[8], Pos=[test.proto:3:15] called from fullIdent.go:15
2020/12/04 21:05:04 [DEBUG] Text=[;], Token=[8], Pos=[test.proto:3:15] called from package.go:55
2020/12/04 21:05:04 [DEBUG] Text=[option], Token=[2], Pos=[test.proto:5:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[option], Token=[2], Pos=[test.proto:5:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[option], Token=[2], Pos=[test.proto:5:1] called from parser.go:44
2020/12/04 21:05:04 [DEBUG] Text=[option], Token=[30], Pos=[test.proto:5:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[option], Token=[30], Pos=[test.proto:5:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[java_package], Token=[2], Pos=[test.proto:5:8] called from option.go:153
2020/12/04 21:05:04 [DEBUG] Text=[=], Token=[10], Pos=[test.proto:5:21] called from option.go:175
2020/12/04 21:05:04 [DEBUG] Text=[=], Token=[10], Pos=[test.proto:5:21] called from option.go:58
2020/12/04 21:05:04 [DEBUG] Text=["], Token=[11], Pos=[test.proto:5:23] called from lexer.go:154
2020/12/04 21:05:04 [DEBUG] Text=["com.test.foo.bar"], Token=[6], Pos=[test.proto:5:23] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[;], Token=[8], Pos=[test.proto:5:41] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[;], Token=[8], Pos=[test.proto:5:41] called from option.go:68
2020/12/04 21:05:04 [DEBUG] Text=[message], Token=[2], Pos=[test.proto:7:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[message], Token=[2], Pos=[test.proto:7:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[message], Token=[2], Pos=[test.proto:7:1] called from parser.go:44
2020/12/04 21:05:04 [DEBUG] Text=[message], Token=[26], Pos=[test.proto:7:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[message], Token=[26], Pos=[test.proto:7:1] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[TestMessage], Token=[2], Pos=[test.proto:7:9] called from message.go:76
2020/12/04 21:05:04 [DEBUG] Text=[{], Token=[14], Pos=[test.proto:7:21] called from message.go:106
2020/12/04 21:05:04 [DEBUG] Text=[string], Token=[2], Pos=[test.proto:7:23] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[string], Token=[2], Pos=[test.proto:7:23] called from message.go:114
2020/12/04 21:05:04 [DEBUG] Text=[string], Token=[2], Pos=[test.proto:7:23] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[string], Token=[2], Pos=[test.proto:7:23] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[string], Token=[2], Pos=[test.proto:7:23] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[string], Token=[2], Pos=[test.proto:7:23] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[string], Token=[2], Pos=[test.proto:7:23] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[string], Token=[2], Pos=[test.proto:7:23] called from field.go:210
2020/12/04 21:05:04 [DEBUG] Text=[test_field], Token=[2], Pos=[test.proto:7:30] called from field.go:79
2020/12/04 21:05:04 [DEBUG] Text=[=], Token=[10], Pos=[test.proto:7:41] called from field.go:85
2020/12/04 21:05:04 [DEBUG] Text=[1], Token=[3], Pos=[test.proto:7:43] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[;], Token=[8], Pos=[test.proto:7:44] called from field.go:119
2020/12/04 21:05:04 [DEBUG] Text=[;], Token=[8], Pos=[test.proto:7:44] called from field.go:100
2020/12/04 21:05:04 [DEBUG] Text=[}], Token=[15], Pos=[test.proto:7:46] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[}], Token=[15], Pos=[test.proto:7:46] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[}], Token=[15], Pos=[test.proto:7:46] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[}], Token=[15], Pos=[test.proto:7:46] called from message.go:151
2020/12/04 21:05:04 [DEBUG] Text=[], Token=[1], Pos=[test.proto:7:47] called from lexer.go:189
2020/12/04 21:05:04 [DEBUG] Text=[], Token=[1], Pos=[test.proto:7:47] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[], Token=[1], Pos=[test.proto:7:47] called from lexer.go:139
2020/12/04 21:05:04 [DEBUG] Text=[], Token=[1], Pos=[test.proto:7:47] called from parser.go:44
[test.proto:7:23] Found an incorrect indentation style "   ". "  " is correct.
```

**Comment by yoheimuta** [comment]
@navaneeth-spotnana Thank you for reporting a bug.
The current implementation seems to have the assumption that a field of the message is at the next line.
This should be fixed.

**Comment by navaneeth-spotnana** [comment]
I observed similar behaviour with return statements:

![image](https://user-images.githubusercontent.com/57885431/101237791-86250480-3701-11eb-809c-b92bc07a3e02.png)

```
syntax = "proto3";

package foobar;

option java_package = "com.test.foo.bar";

message TestMessage { string test_field = 1; }

service FooBarService {
  rpc Put(TestMessage)
    returns (TestMessage) {}
}
```

Are you going to take this up and fix it? Golang is not very familiar for me, I can take a stab at it otherwise.

**Comment by navaneeth-spotnana** [comment]
I was reading the code, and this is what I understand:

```
func (v indentVisitor) validateIndent(
	pos meta.Position,
) {
	line := v.protoLines[pos.Line-1]
	leading := ""
	for _, r := range string([]rune(line)[0 : pos.Column-1]) {
		if unicode.IsSpace(r) {
			leading += string(r)
		}
	}

	indentation := strings.Repeat(v.style, v.currentLevel)
	if leading == indentation {
		return
	}
	v.AddFailuref(
		pos,
		`Found an incorrect indentation style "%s". "%s" is correct.`,
		leading,
		indentation,
	)

	if v.fixMode {
		v.indentFixes[pos.Line-1] = indentFix{
			currentChars: len(leading),
			replacement:  indentation,
		}
	}
}
```

The above code snippet won't work in case there are more than one elements on the same line. What would be the fix for this?

**Comment by yoheimuta** [comment]
@navaneeth-spotnana Thank you for investigating this issue!
That's right. First, I think it's necessary to decide the solution users can satisfy.

What do you think about the change that protolint will do nothing against inner elements on the same line? 

PS. If someone wants to format a file like converting to multiple lines with correct indentation, it would be better to support a new `format` command which rewrites the file completely according to the opinionated rule.

**Comment by navaneeth-spotnana** [comment]
> What do you think about the change that protolint will do nothing against inner elements on the same line?
👎🏽 Please don't. That's very important for me. I am considering using protolint inside our company, and if it doesn't insert proper linebreaks, it will be disappointing.

> PS. If someone wants to format a file like converting to multiple lines with correct indentation, it would be better to support a new `format` command which rewrites the file completely according to the opinionated rule.

Agreed, I had the same thoughts while going through the code last week. The proto file is already parsed into the json form. Full rewrite might be easier than editing. Unless you add support for newlines.

**Comment by yoheimuta** [comment]
@navaneeth-spotnana Thank you for sharing your thought.
I'm working on this issue at the moment. It's almost done, so wait a while.

**Comment by daltonfury42** [comment]
In a way, full rewrite might be better, as this might enable us to support [file structure ](https://developers.google.com/protocol-buffers/docs/style#file_structure)guidelines?

**Comment by yoheimuta** [comment]
@daltonfury42 That's right.
I created an issue https://github.com/yoheimuta/protolint/issues/142 for the moment.

**Comment by yoheimuta** [comment]
@navaneeth-spotnana I fixed this issue by #141.
It also features inserting proper linebreaks by default.

Can you confirm [v0.27.0](https://github.com/yoheimuta/protolint/releases/tag/v0.27.0) works for you?

**Comment by navaneeth-spotnana** [comment]
Yes it's working, thanks a lot.
